src/_gettsim/transfers/arbeitsl_geld_2/sgb_ii_bedarf.py

```diff
@@ -0,0 +1,275 @@
+    you have at least one kid below 6 or two or three below 15, you get 36%
+    on top alternatively, you get 12% per kid, depending on what's higher.
+
+    Note: Since 2023, Arbeitslosengeld 2 is referred to as Bürgergeld.
+
+    Parameters
+    ----------
+    alleinerz
+        See :func:`alleinerz`.
+    anz_kinder_fg
+        See :func:`anz_kinder_fg`.
+    anz_kinder_bis_6_fg
+        See :func:`anz_kinder_bis_6_fg`.
+    anz_kinder_bis_15_fg
+        See :func:`anz_kinder_bis_15_fg`.
+    arbeitsl_geld_2_params
+        See params documentation :ref:`arbeitsl_geld_2_params <arbeitsl_geld_2_params>`.
+
+
+    Returns
+    -------
+    float checks how much more a single parent need.
+
+    """
+    if alleinerz:
+        # Clip value at calculated minimal share and given upper share
+        # Note that upper limit is applied last (for many children lower
+        # could be greater than upper)
+        out = min(
+            max(
+                # Minimal Mehrbedarf share. Minimal rate times number of children
+                arbeitsl_geld_2_params["mehrbedarf_anteil"]["min_1_kind"]
+                * anz_kinder_fg,
+                # Special case if 1 kid below 6 or 2,3 below 15.
+                (
+                    arbeitsl_geld_2_params["mehrbedarf_anteil"][
+                        "kind_unter_7_oder_mehr"
+                    ]
+                    if (anz_kinder_bis_6_fg >= 1) or (2 <= anz_kinder_bis_15_fg <= 3)
+                    else 0.0
+                ),
+            ),
+            arbeitsl_geld_2_params["mehrbedarf_anteil"]["max"],
+        )
+    else:
+        out = 0.0
+    return out
+
+
+@policy_info(end_date="2010-12-31", name_in_dag="arbeitsl_geld_2_kindersatz_m")
+def arbeitsl_geld_2_kindersatz_m_bis_2010(
+    alter: int,
+    gleiche_fg_kindergeldempfänger_kind: bool,
+    arbeitsl_geld_2_params: dict,
+) -> float:
+    """Basic monthly subsistence / SGB II needs of children until 2010.
+
+    Since 2010 children get additional shares instead of lump sum payments.
+
+    Parameters
+    ----------
+    alter
+        See basic input variable :ref:`alter`.
+    gleiche_fg_kindergeldempfänger_kind
+        See :func:`gleiche_fg_kindergeldempfänger_kind`.
+    arbeitsl_geld_2_params
+        See params documentation :ref:`arbeitsl_geld_2_params <arbeitsl_geld_2_params>`.
+
+    Returns
+    -------
+    float with SGB II needs of children until year 2010.
+
+    """
+    anteile = arbeitsl_geld_2_params["anteil_regelsatz_kinder"]
+    regelsatz = arbeitsl_geld_2_params["regelsatz"]
+
+    if alter <= anteile[6]["max_alter"] and gleiche_fg_kindergeldempfänger_kind:
+        out = regelsatz * anteile[6]["anteil"]
+    elif alter <= anteile[5]["max_alter"] and gleiche_fg_kindergeldempfänger_kind:
+        out = regelsatz * anteile[5]["anteil"]
+    elif alter <= anteile[4]["max_alter"] and gleiche_fg_kindergeldempfänger_kind:
+        out = regelsatz * anteile[4]["anteil"]
+    else:
```

Comment:
Just to double-check: Is it correct that level 3 is missing here but present in 2011+?

Reply:
This whole level thing didn't exist before 2011 and I guess the values before 2011 were chosen to reflect the levels after the 2011 (in both cases, level 6, for example, refers to children younger than 6). The post-2011 level 3 didn't exist back then. Probably this is bad style and we should stick to something that doesn't try a fuzzy category matching before and after the reform.

Reply:
Yes, please check some old law and adjust accordingly. This made sense back in the pre-dates_active days, but not anymore.